apps/system/utils/utils_free.c

```diff
@@ -56,22 +56,48 @@
 #include <tinyara/config.h>
 #include <stdio.h>
 #include <stdlib.h>
+#if defined(CONFIG_BUILD_PROTECTED) && defined(CONFIG_MM_KERNEL_HEAP)
+#include <fcntl.h>
+#include <errno.h>
+
+#include <tinyara/fs/ioctl.h>
+#include <tinyara/meminfo_drv.h>
+#endif
 
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
 int utils_free(int argc, char **args)
 {
+#if defined(CONFIG_BUILD_PROTECTED) && defined(CONFIG_MM_KERNEL_HEAP)
+	int fd;
+	int ret;
+#endif
 	struct mallinfo data;
 
+#if defined(CONFIG_BUILD_PROTECTED)
+	fd = open(MEMINFO_DRVPATH, O_RDWR);
+	if (fd < 0) {
+		printf("Open %s Fail, errno %d\n", MEMINFO_DRVPATH, get_errno());
+		return ERROR;
+	}
+
+	ret = ioctl(fd, MEMINFOIOC_KHEAP, (int)&data);
+	if (ret == ERROR) {
```

Comment:
Close first and use if state more explicitly.
```
close(fd);
if (ret != OK) {
   printf();
   return ERROR;
}
```

Reply:
It is good for safety but we can't know the errno returned from ioctl.